components/brave_shields/browser/ad_block_service.cc

```diff
@@ -62,23 +62,29 @@ AdBlockService::SourceProviderObserver::SourceProviderObserver(
     base::WeakPtr<AdBlockEngine> adblock_engine,
     AdBlockFiltersProvider* filters_provider,
     AdBlockResourceProvider* resource_provider,
-    scoped_refptr<base::SequencedTaskRunner> task_runner,
-    base::RepeatingCallback<void(const adblock::FilterListMetadata&)>
-        on_metadata_retrieved)
+    scoped_refptr<base::SequencedTaskRunner> task_runner)
     : adblock_engine_(adblock_engine),
       filters_provider_(filters_provider),
       resource_provider_(resource_provider),
-      on_metadata_retrieved_(on_metadata_retrieved),
       task_runner_(task_runner) {
   filters_provider_->AddObserver(this);
-  filters_provider_->LoadDAT(this);
+  filters_provider_->LoadDAT(
+      base::BindOnce(&AdBlockService::SourceProviderObserver::OnDATLoaded,
+                     weak_factory_.GetWeakPtr()));
 }
 
 AdBlockService::SourceProviderObserver::~SourceProviderObserver() {
   filters_provider_->RemoveObserver(this);
   resource_provider_->RemoveObserver(this);
 }
 
+void AdBlockService::SourceProviderObserver::OnNewSourceAvailable(
+    base::WeakPtr<AdBlockFiltersProvider> from) {
+  from->LoadDAT(
```

Comment:
WeakPtr does nothing here if you don't check to see if `from` is null. Also nit I would change from -> filters_provider so it's more descriptive

Reply:
I'm actually going to remove the `from` argument completely. We already have access to `filters_provider_` here. I thought it would be useful for `AdBlockCombinatorFiltersProvider` to know which list was updated, but it turns out that's not important either.